From 5f3ec81229f4f94ec89b79cd34bcdd8d544fcfe0 Mon Sep 17 00:00:00 2001
From: Irina Patru <irina.patru@intel.com>
Date: Wed, 19 Aug 2015 16:33:58 +0300
Subject: [PATCH 09/17] input: silead: Fix GSL1680 reporting points

GSL1680 touch was wrongly detected with GSL1688_CHIP_ID definition,
which needed to be removed in order to preserve GSL1680 functionality.

Change-Id: I9a6b6b9f392a30b8a35c3d2da72f3c619147a5ff
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Irina Patru <irina.patru@intel.com>
Reviewed-on: https://android.intel.com:443/472782
---
 drivers/input/touchscreen/silead.c | 35 ++++++++++++++---------------------
 1 file changed, 14 insertions(+), 21 deletions(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 3a5cbe4..3c3afb3 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -78,8 +78,6 @@
 #define SILEAD_PRESSURE		50
 #define SILEAD_FW_NAME_LEN	30
 
-#define GSL1688_CHIP_ID		0xB4820000
-
 enum silead_ts_power {
 	SILEAD_POWER_ON  = 1,
 	SILEAD_POWER_OFF = 0
@@ -207,25 +205,20 @@ static void silead_ts_read_data(struct i2c_client *client)
 
 		dev_dbg(dev, "x=%d y=%d id=%d\n", x, y, id);
 
-		if (data->chip_id == GSL1688_CHIP_ID)
-			silead_ts_report_touch(data, (id * 256 + x),
-					       data->y_max - y, id);
-		else {
-			if (data->xy_swap)
-				silead_ts_report_touch(data,
-						       data->y_invert ?
-						       data->y_max - y : y,
-						       data->x_invert ?
-						       data->x_max - x : x,
-						       id);
-			else
-				silead_ts_report_touch(data,
-						       data->x_invert ?
-						       data->x_max - x : x,
-						       data->y_invert ?
-						       data->y_max - y : y,
-						       id);
-		}
+		if (data->xy_swap)
+			silead_ts_report_touch(data,
+					       data->y_invert ?
+					       data->y_max - y : y,
+					       data->x_invert ?
+					       data->x_max - x : x,
+					       id);
+		else
+			silead_ts_report_touch(data,
+					       data->x_invert ?
+					       data->x_max - x : x,
+					       data->y_invert ?
+					       data->y_max - y : y,
+					       id);
 	}
 
 	input_mt_sync_frame(data->input_dev);
-- 
1.9.1

